libc/test/src/sys/ioctl/linux/ioctl_test.cpp

```diff
@@ -0,0 +1,36 @@
+//===-- Unittests for ioctl -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/__support/OSUtil/syscall.h" // For internal syscall function.
+#include "src/errno/libc_errno.h"
+#include "src/sys/ioctl/ioctl.h"
+#include "test/UnitTest/ErrnoSetterMatcher.h"
+#include "test/UnitTest/LibcTest.h"
+#include "test/UnitTest/Test.h"
+
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+using LIBC_NAMESPACE::testing::ErrnoSetterMatcher::Fails;
+using LIBC_NAMESPACE::testing::ErrnoSetterMatcher::Succeeds;
+
+TEST(LlvmLibcIoctlTest, InvalidFileDescriptor) {
+  int fd = 10;
+  unsigned long request = 10;
+  int res = LIBC_NAMESPACE::ioctl(fd, request, NULL);
+  EXPECT_THAT(res, Fails(EBADF, -1));
+}
+
+TEST(LlvmLibcIoctlTest, ValidFileDescriptor) {
+  int fd = open("/dev/null", O_RDWR);
+  int data;
+  int res = LIBC_NAMESPACE::ioctl(fd, FIONREAD, &data);
+  EXPECT_THAT(res, Succeeds());
```

Comment:
From the presubmit build failure:
```
ioctl_test.cpp:35: FAILURE
Failed to match res against Succeeds().
Expected return value to be equal to 0 but got -1.
Expected errno to be equal to "Success" but got "Inappropriate ioctl for device".
```

Reply:
I applied `FS_IOC_GETFLAGS ` but it seems the same error is still there

Reply:
it looks like the error `In appropriate ioctl for device` is for the case the test wants to connect to terminal but there is no terminal on the CI.

Reply:
This is perhaps related to the usage of `/dev/null` as the input file.  Perhaps we need to create a new file and run the test on that.

Reply:
```sh
$ lsattr /dev/null
lsattr: Operation not supported While reading flags on /dev/null
```

Reply:
See how libc/test/src/stdio/ tests create temp files.

Reply:
looks like it works now

Reply:
make sure to close the file

Reply:
@nickdesaulniers nick
do u know why I am getting error undefined symbol to libc_namspace::open() and close()?

Reply:
Is that a linkage failure? Then that would be a missing dependency in cmake for the test.